W0424 12:56:03.815285 2244338 site-packages/torch/_dynamo/variables/tensor.py:867] [0/0] Graph break from `Tensor.item()`, consider setting:
W0424 12:56:03.815285 2244338 site-packages/torch/_dynamo/variables/tensor.py:867] [0/0]     torch._dynamo.config.capture_scalar_outputs = True
W0424 12:56:03.815285 2244338 site-packages/torch/_dynamo/variables/tensor.py:867] [0/0] or:
W0424 12:56:03.815285 2244338 site-packages/torch/_dynamo/variables/tensor.py:867] [0/0]     env TORCHDYNAMO_CAPTURE_SCALAR_OUTPUTS=1
W0424 12:56:03.815285 2244338 site-packages/torch/_dynamo/variables/tensor.py:867] [0/0] to include these operations in the captured graph.
W0424 12:56:03.815285 2244338 site-packages/torch/_dynamo/variables/tensor.py:867] [0/0]
W0424 12:56:03.815285 2244338 site-packages/torch/_dynamo/variables/tensor.py:867] [0/0] Graph break: from user code at:
W0424 12:56:03.815285 2244338 site-packages/torch/_dynamo/variables/tensor.py:867] [0/0]   File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/external_utils.py", line 31, in inner
W0424 12:56:03.815285 2244338 site-packages/torch/_dynamo/variables/tensor.py:867] [0/0]     return fn(*args, **kwargs)
W0424 12:56:03.815285 2244338 site-packages/torch/_dynamo/variables/tensor.py:867] [0/0]   File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/utils/_contextlib.py", line 116, in decorate_context
W0424 12:56:03.815285 2244338 site-packages/torch/_dynamo/variables/tensor.py:867] [0/0]     return func(*args, **kwargs)
W0424 12:56:03.815285 2244338 site-packages/torch/_dynamo/variables/tensor.py:867] [0/0]   File "/home/levi/Desktop/MOORE-TDMPC/TDMPC2/algorithms/tdmpc2.py", line 168, in _plan
W0424 12:56:03.815285 2244338 site-packages/torch/_dynamo/variables/tensor.py:867] [0/0]     z = self.model.encode(obs, task)
W0424 12:56:03.815285 2244338 site-packages/torch/_dynamo/variables/tensor.py:867] [0/0]   File "/home/levi/Desktop/MOORE-TDMPC/moore_tdmpc/train_moore_agent.py", line 249, in encode_only
W0424 12:56:03.815285 2244338 site-packages/torch/_dynamo/variables/tensor.py:867] [0/0]     out = orig_encode(obs, task) if task is not None else orig_encode(obs)
W0424 12:56:03.815285 2244338 site-packages/torch/_dynamo/variables/tensor.py:867] [0/0]   File "/home/levi/Desktop/MOORE-TDMPC/moore_tdmpc/world_model.py", line 228, in encode
W0424 12:56:03.815285 2244338 site-packages/torch/_dynamo/variables/tensor.py:867] [0/0]     f = int(torch.tensor(x.shape[1:]).prod().item())
W0424 12:56:03.815285 2244338 site-packages/torch/_dynamo/variables/tensor.py:867] [0/0]
W0424 12:56:03.815285 2244338 site-packages/torch/_dynamo/variables/tensor.py:867] [0/0]
W0424 12:56:06.081346 2244338 site-packages/torch/_logging/_internal.py:1084] [15/0] Profiler function <class 'torch.autograd.profiler.record_function'> will be ignored
[34m[1mLogs will be synced with wandb.
Config(env_type='MooreMultiTask', n_contexts=10, obs_dim=39, action_dim=4, hidden_dim=256, n_experts=4, moore_temperature=1.0, use_softmax=True, use_checkpoint=False, use_mixed_precision=True, gamma=0.99, stddev=0.1, actor_updates=2, use_ema=True, ensemble_size=5, update_freq=2, seed=42, device=device(type='cuda', index=0), task='mw-hammer', obs='state', multitask=False, steps=10000000, batch_size=1024, reward_coef=0.1, value_coef=0.25, consistency_coef=20, rho=0.5, lr=0.0003, enc_lr_scale=0.3, grad_clip_norm=20, tau=0.01, discount_denom=5, discount_min=0.95, discount_max=0.995, buffer_size=100000, exp_name='moore-tdmpc2-mt30', eval_freq=5000, eval_episodes=10, moore={'n_experts': 4, 'temperature': 1.0, 'use_softmax': True, 'debug_task_emb': False}, mpc=True, iterations=4, num_samples=512, num_elites=64, num_pi_trajs=24, horizon=5, min_std=0.05, max_std=2, temperature=0.5, log_std_min=-10, log_std_max=2, entropy_coef=0.0001, model_size=5, num_enc_layers=2, enc_dim=256, num_channels=32, mlp_dim=512, latent_dim=512, task_dim=0, num_q=5, dropout=0.01, simnorm_dim=8, num_bins=101, vmin=-10, vmax=10, wandb_project='moore-tdmpc2', wandb_entity='OA-MBRL', wandb_silent=False, enable_wandb=True, save_csv=True, save_video=False, save_agent=True, compile=True, compile_dynamics=True, compile_loss=True, work_dir=PosixPath('/home/levi/Desktop/MOORE-TDMPC/moore_tdmpc/logs/mw-hammer/42/moore-tdmpc2-mt30'), task_title='Mw Hammer', tasks=['mw-hammer'], obs_shape={'state': (39,)}, episode_length=100, obs_shapes='???', episode_lengths='???', seed_steps=1000, bin_size=0.2, data_dir='/media/levi/Singe4linux/Moore-TDMPC/TDMPC2/data/mt30')
Architecture: MooreWorldModel(total_params=5308978, latent_dim=512, action_dim=4, n_experts=4, temperature=0.5)
Task Encoder: MooreTaskEncoder(obs_dim=39, latent_dim=512, hidden_dim=256, n_experts=4, temperature=1.0, use_softmax=True)
Transition-Reward Core: MoETransitionRewardModel(latent_dim=512, action_dim=4, hidden_dim=256, n_experts=4, temperature=0.5, use_softmax=True, reward_dim=101, fused_head=True, use_mixed_precision=True)
Policy prior: Sequential(
  (0): Linear(in_features=512, out_features=256, bias=True)
  (1): ReLU()
  (2): Linear(in_features=256, out_features=256, bias=True)
  (3): ReLU()
  (4): Linear(in_features=256, out_features=8, bias=True)
)
Q-functions: Ensemble(n_models=5)
[32m[1må¼€å§‹Moore-TDMPC2è®­ç»ƒ...
[32mOnlineTrainer: Starting training loop with profiling enabled
[36mPROFILER STATUS: Enabled=True, Functions=0, Total Calls=0
[36mPROFILER: forward executed in 589.18ms (call 1)
[36mPROFILER: forward executed in 248.91ms (call 2)
[36mPROFILER: forward executed in 13.35ms (call 3)
W0424 12:56:22.239691 2244338 site-packages/torch/_dynamo/convert_frame.py:897] [6/8] torch._dynamo hit config.cache_size_limit (8)
W0424 12:56:22.239691 2244338 site-packages/torch/_dynamo/convert_frame.py:897] [6/8]    function: 'torch_dynamo_resume_in_wrapper_at_27' (/home/levi/Desktop/MOORE-TDMPC/moore_tdmpc/utils/profiler.py:27)
W0424 12:56:22.239691 2244338 site-packages/torch/_dynamo/convert_frame.py:897] [6/8]    last reason: 6/0: L['___stack0'] == 1745470565.392921
W0424 12:56:22.239691 2244338 site-packages/torch/_dynamo/convert_frame.py:897] [6/8] To log all recompilation reasons, use TORCH_LOGS="recompiles".
W0424 12:56:22.239691 2244338 site-packages/torch/_dynamo/convert_frame.py:897] [6/8] To diagnose recompilation issues, see https://pytorch.org/docs/main/torch.compiler_troubleshooting.html.
=== é¦–æ¬¡ eval() é˜¶æ®µçƒ­ç‚¹å‡½æ•° ===
         55218867 function calls (52668441 primitive calls) in 23.720 seconds
   Ordered by: cumulative time
   List reduced from 4987 to 20 due to restriction <20>
   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
        1    0.013    0.013   24.347   24.347 /home/levi/Desktop/MOORE-TDMPC/TDMPC2/algorithms/trainer/online_trainer.py:36(eval)
2161/1000    0.014    0.000   23.333    0.023 /home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/utils/_contextlib.py:113(decorate_context)
     1000    0.003    0.000   23.321    0.023 /home/levi/Desktop/MOORE-TDMPC/TDMPC2/algorithms/tdmpc2.py:92(act)
     1000    0.013    0.000   19.344    0.019 /home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/eval_frame.py:525(_fn)
     1000    0.048    0.000   19.227    0.019 /home/levi/Desktop/MOORE-TDMPC/TDMPC2/algorithms/tdmpc2.py:153(_plan)
       72    0.001    0.000   17.828    0.248 /home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/convert_frame.py:1327(__call__)
       53    0.000    0.000   17.823    0.336 /home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/convert_frame.py:1164(__call__)
       53    0.003    0.000   17.822    0.336 /home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/convert_frame.py:447(__call__)
       31    0.001    0.000   17.817    0.575 /home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/convert_frame.py:596(_compile)
       30    0.000    0.000   17.798    0.593 /home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/convert_frame.py:688(compile_inner)
       30    0.000    0.000   17.795    0.593 /home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_utils_internal.py:89(wrapper_function)
       30    0.001    0.000   17.795    0.593 /home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/convert_frame.py:710(_compile_inner)
    68/52    0.000    0.000   17.519    0.337 /home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/bytecode_transformation.py:1340(transform_code_object)
       52    0.001    0.000   17.485    0.336 /home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/convert_frame.py:202(_fn)
       52    0.001    0.000   17.479    0.336 /home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/convert_frame.py:631(transform)
       52    0.000    0.000   17.449    0.336 /home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/symbolic_convert.py:2911(run)
  1389/52    0.009    0.000   17.449    0.336 /home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/symbolic_convert.py:1116(run)
35102/3088    0.047    0.000   17.445    0.006 /home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/symbolic_convert.py:1004(step)
       26    0.001    0.000   13.997    0.538 /home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/output_graph.py:937(compile_subgraph)
        6    0.000    0.000   13.977    2.329 /home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/output_graph.py:1268(compile_and_call_fx_graph)
 [32meval[39m    [34mE:[39m 0            [34mI:[39m 0            [34mR:[39m 143.8        [34mS:[39m 0.3          [34mT:[39m 0:00:24
 [34mtrain[39m   [34mE:[39m 0            [34mI:[39m 100          [34mR:[39m 184.5        [34mS:[39m 0.0          [34mT:[39m 0:00:24
Buffer capacity: 100,000
Storage required: 0.02 GB
Using CUDA:0 memory for storage.
 [34mtrain[39m   [34mE:[39m 1            [34mI:[39m 200          [34mR:[39m 236.8        [34mS:[39m 0.0          [34mT:[39m 0:00:24
 [34mtrain[39m   [34mE:[39m 2            [34mI:[39m 300          [34mR:[39m 170.3        [34mS:[39m 0.0          [34mT:[39m 0:00:24
 [34mtrain[39m   [34mE:[39m 3            [34mI:[39m 400          [34mR:[39m 172.7        [34mS:[39m 0.0          [34mT:[39m 0:00:24
 [34mtrain[39m   [34mE:[39m 4            [34mI:[39m 500          [34mR:[39m 182.8        [34mS:[39m 0.0          [34mT:[39m 0:00:24
 [34mtrain[39m   [34mE:[39m 5            [34mI:[39m 600          [34mR:[39m 186.2        [34mS:[39m 0.0          [34mT:[39m 0:00:24
 [34mtrain[39m   [34mE:[39m 6            [34mI:[39m 700          [34mR:[39m 176.2        [34mS:[39m 0.0          [34mT:[39m 0:00:24
E0424 12:56:28.995204 2244338 site-packages/torch/_subclasses/fake_tensor.py:2301] [18/0] failed while attempting to run meta for aten.sub.Tensor
E0424 12:56:28.995204 2244338 site-packages/torch/_subclasses/fake_tensor.py:2301] [18/0] Traceback (most recent call last):
E0424 12:56:28.995204 2244338 site-packages/torch/_subclasses/fake_tensor.py:2301] [18/0]   File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_subclasses/fake_tensor.py", line 2297, in _dispatch_impl
E0424 12:56:28.995204 2244338 site-packages/torch/_subclasses/fake_tensor.py:2301] [18/0]     r = func(*args, **kwargs)
E0424 12:56:28.995204 2244338 site-packages/torch/_subclasses/fake_tensor.py:2301] [18/0]   File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_ops.py", line 723, in __call__
E0424 12:56:28.995204 2244338 site-packages/torch/_subclasses/fake_tensor.py:2301] [18/0]     return self._op(*args, **kwargs)
E0424 12:56:28.995204 2244338 site-packages/torch/_subclasses/fake_tensor.py:2301] [18/0]   File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_prims_common/wrappers.py", line 291, in _fn
E0424 12:56:28.995204 2244338 site-packages/torch/_subclasses/fake_tensor.py:2301] [18/0]     result = fn(*args, **kwargs)
E0424 12:56:28.995204 2244338 site-packages/torch/_subclasses/fake_tensor.py:2301] [18/0]   File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_prims_common/wrappers.py", line 143, in _fn
E0424 12:56:28.995204 2244338 site-packages/torch/_subclasses/fake_tensor.py:2301] [18/0]     result = fn(**bound.arguments)
E0424 12:56:28.995204 2244338 site-packages/torch/_subclasses/fake_tensor.py:2301] [18/0]   File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_refs/__init__.py", line 1732, in sub
E0424 12:56:28.995204 2244338 site-packages/torch/_subclasses/fake_tensor.py:2301] [18/0]     a, b = _maybe_broadcast(a, b)
E0424 12:56:28.995204 2244338 site-packages/torch/_subclasses/fake_tensor.py:2301] [18/0]   File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_refs/__init__.py", line 437, in _maybe_broadcast
E0424 12:56:28.995204 2244338 site-packages/torch/_subclasses/fake_tensor.py:2301] [18/0]     common_shape = _broadcast_shapes(
E0424 12:56:28.995204 2244338 site-packages/torch/_subclasses/fake_tensor.py:2301] [18/0]   File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_refs/__init__.py", line 425, in _broadcast_shapes
E0424 12:56:28.995204 2244338 site-packages/torch/_subclasses/fake_tensor.py:2301] [18/0]     torch._check(
E0424 12:56:28.995204 2244338 site-packages/torch/_subclasses/fake_tensor.py:2301] [18/0]   File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/__init__.py", line 1617, in _check
E0424 12:56:28.995204 2244338 site-packages/torch/_subclasses/fake_tensor.py:2301] [18/0]     _check_with(RuntimeError, cond, message)
E0424 12:56:28.995204 2244338 site-packages/torch/_subclasses/fake_tensor.py:2301] [18/0]   File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/__init__.py", line 1599, in _check_with
E0424 12:56:28.995204 2244338 site-packages/torch/_subclasses/fake_tensor.py:2301] [18/0]     raise error_type(message_evaluated)
E0424 12:56:28.995204 2244338 site-packages/torch/_subclasses/fake_tensor.py:2301] [18/0] RuntimeError: Attempting to broadcast a dimension of length 6144 at -2! Mismatching argument at index 1 had torch.Size([6144, 4]); but expected shape should be broadcastable to [5120, 4]
Error executing job with overrides: ['task=mw-hammer']
Traceback (most recent call last):
  File "/home/levi/Desktop/MOORE-TDMPC/moore_tdmpc/train_with_cprofile.py", line 161, in train
    trainer.train()
  File "/home/levi/Desktop/MOORE-TDMPC/moore_tdmpc/utils/profiler.py", line 28, in wrapper
    result = func(*args, **kwargs)
  File "/home/levi/Desktop/MOORE-TDMPC/TDMPC2/algorithms/trainer/online_trainer.py", line 126, in train
    _train_metrics = self.agent.update(self.buffer)
  File "/home/levi/Desktop/MOORE-TDMPC/moore_tdmpc/train_moore_agent.py", line 500, in update
    return self._update(obs, action, reward, task=task)
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/eval_frame.py", line 556, in _fn
    return fn(*args, **kwargs)
  File "/home/levi/Desktop/MOORE-TDMPC/moore_tdmpc/train_moore_agent.py", line 293, in _update
    z_seq = normalize_bt(self.model.encode(obs, task), 'obs')
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/convert_frame.py", line 1395, in __call__
    return self._torchdynamo_orig_callable(
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/convert_frame.py", line 1180, in __call__
    result = self._inner_convert(
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/convert_frame.py", line 545, in __call__
    return _compile(
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/convert_frame.py", line 977, in _compile
    guarded_code = compile_inner(code, one_graph, hooks, transform)
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/convert_frame.py", line 706, in compile_inner
    return _compile_inner(code, one_graph, hooks, transform)
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_utils_internal.py", line 95, in wrapper_function
    return function(*args, **kwargs)
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/convert_frame.py", line 741, in _compile_inner
    out_code = transform_code_object(code, transform)
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/bytecode_transformation.py", line 1348, in transform_code_object
    transformations(instructions, code_options)
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/convert_frame.py", line 229, in _fn
    return fn(*args, **kwargs)
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/convert_frame.py", line 658, in transform
    tracer.run()
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/symbolic_convert.py", line 2912, in run
    super().run()
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/symbolic_convert.py", line 1120, in run
    while self.step():
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/symbolic_convert.py", line 1032, in step
    self.dispatch_table[inst.opcode](self, inst)
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/symbolic_convert.py", line 640, in wrapper
    return inner_fn(self, inst)
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/symbolic_convert.py", line 1828, in CALL_FUNCTION_KW
    self.call_function(fn, args, kwargs)
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/symbolic_convert.py", line 967, in call_function
    self.push(fn.call_function(self, args, kwargs))  # type: ignore[arg-type]
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/variables/functions.py", line 410, in call_function
    return super().call_function(tx, args, kwargs)
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/variables/functions.py", line 349, in call_function
    return super().call_function(tx, args, kwargs)
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/variables/functions.py", line 125, in call_function
    return tx.inline_user_function_return(self, [*self.self_args(), *args], kwargs)
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/symbolic_convert.py", line 973, in inline_user_function_return
    return InliningInstructionTranslator.inline_call(self, fn, args, kwargs)
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/symbolic_convert.py", line 3127, in inline_call
    return cls.inline_call_(parent, func, args, kwargs)
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/symbolic_convert.py", line 3255, in inline_call_
    tracer.run()
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/symbolic_convert.py", line 1120, in run
    while self.step():
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/symbolic_convert.py", line 1032, in step
    self.dispatch_table[inst.opcode](self, inst)
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/symbolic_convert.py", line 640, in wrapper
    return inner_fn(self, inst)
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/symbolic_convert.py", line 1738, in CALL_FUNCTION
    self.call_function(fn, args, {})
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/symbolic_convert.py", line 967, in call_function
    self.push(fn.call_function(self, args, kwargs))  # type: ignore[arg-type]
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/variables/functions.py", line 410, in call_function
    return super().call_function(tx, args, kwargs)
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/variables/functions.py", line 349, in call_function
    return super().call_function(tx, args, kwargs)
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/variables/functions.py", line 125, in call_function
    return tx.inline_user_function_return(self, [*self.self_args(), *args], kwargs)
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/symbolic_convert.py", line 973, in inline_user_function_return
    return InliningInstructionTranslator.inline_call(self, fn, args, kwargs)
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/symbolic_convert.py", line 3127, in inline_call
    return cls.inline_call_(parent, func, args, kwargs)
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/symbolic_convert.py", line 3255, in inline_call_
    tracer.run()
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/symbolic_convert.py", line 1120, in run
    while self.step():
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/symbolic_convert.py", line 1032, in step
    self.dispatch_table[inst.opcode](self, inst)
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/symbolic_convert.py", line 294, in impl
    self.push(fn_var.call_function(self, self.popn(nargs), {}))
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/variables/builtin.py", line 1004, in call_function
    return handler(tx, args, kwargs)
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/variables/builtin.py", line 980, in _handle_insert_op_in_graph
    return wrap_fx_proxy(tx, proxy)
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/variables/builder.py", line 2108, in wrap_fx_proxy
    return wrap_fx_proxy_cls(target_cls=TensorVariable, **kwargs)
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/variables/builder.py", line 2174, in wrap_fx_proxy_cls
    return _wrap_fx_proxy(
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/variables/builder.py", line 2270, in _wrap_fx_proxy
    example_value = get_fake_value(proxy.node, tx, allow_non_graph_fake=True)
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/utils.py", line 2278, in get_fake_value
    raise TorchRuntimeError(str(e)).with_traceback(e.__traceback__) from None
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/utils.py", line 2213, in get_fake_value
    ret_val = wrap_fake_exception(
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/utils.py", line 1761, in wrap_fake_exception
    return fn()
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/utils.py", line 2214, in <lambda>
    lambda: run_node(tx.output, node, args, kwargs, nnmodule)
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/utils.py", line 2346, in run_node
    raise RuntimeError(make_error_message(e)).with_traceback(
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_dynamo/utils.py", line 2328, in run_node
    return node.target(*args, **kwargs)
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/utils/_stats.py", line 21, in wrapper
    return fn(*args, **kwargs)
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_subclasses/fake_tensor.py", line 1271, in __torch_dispatch__
    return self.dispatch(func, types, args, kwargs)
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_subclasses/fake_tensor.py", line 1813, in dispatch
    return self._cached_dispatch_impl(func, types, args, kwargs)
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_subclasses/fake_tensor.py", line 1372, in _cached_dispatch_impl
    output = self._dispatch_impl(func, types, args, kwargs)
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_subclasses/fake_tensor.py", line 2297, in _dispatch_impl
    r = func(*args, **kwargs)
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_ops.py", line 723, in __call__
    return self._op(*args, **kwargs)
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_prims_common/wrappers.py", line 291, in _fn
    result = fn(*args, **kwargs)
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_prims_common/wrappers.py", line 143, in _fn
    result = fn(**bound.arguments)
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_refs/__init__.py", line 1732, in sub
    a, b = _maybe_broadcast(a, b)
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_refs/__init__.py", line 437, in _maybe_broadcast
    common_shape = _broadcast_shapes(
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/_refs/__init__.py", line 425, in _broadcast_shapes
    torch._check(
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/__init__.py", line 1617, in _check
    _check_with(RuntimeError, cond, message)
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/__init__.py", line 1599, in _check_with
    raise error_type(message_evaluated)
torch._dynamo.exc.TorchRuntimeError: Failed running call_function <built-in function sub>(*(FakeTensor(..., device='cuda:0', size=(5120, 4)), FakeTensor(..., device='cuda:0', size=(6144, 4), dtype=torch.float16,
           grad_fn=<SplitBackward0>)), **{}):
Attempting to broadcast a dimension of length 6144 at -2! Mismatching argument at index 1 had torch.Size([6144, 4]); but expected shape should be broadcastable to [5120, 4]
from user code:
   File "/home/levi/Desktop/MOORE-TDMPC/moore_tdmpc/train_moore_agent.py", line 321, in torch_dynamo_resume_in__update_at_293
    pl, rl, vl, cl, tot = self._compiled_loss_and_backward(
  File "/home/levi/Desktop/MOORE-TDMPC/moore_tdmpc/train_moore_agent.py", line 362, in _loss_and_backward
    log_probs = dist.log_prob(a_flat)                # [B*H, A]
  File "/home/levi/anaconda3/envs/tdmpc2/lib/python3.9/site-packages/torch/distributions/normal.py", line 89, in log_prob
    -((value - self.loc) ** 2) / (2 * var)
Set TORCH_LOGS="+dynamo" and TORCHDYNAMO_VERBOSE=1 for more information
You can suppress this exception and fall back to eager by setting:
    import torch._dynamo
    torch._dynamo.config.suppress_errors = True
Set the environment variable HYDRA_FULL_ERROR=1 for a complete stack trace.
 [34mtrain[39m   [34mE:[39m 7            [34mI:[39m 800          [34mR:[39m 231.9        [34mS:[39m 0.0          [34mT:[39m 0:00:24
 [34mtrain[39m   [34mE:[39m 8            [34mI:[39m 900          [34mR:[39m 222.9        [34mS:[39m 0.0          [34mT:[39m 0:00:24
 [34mtrain[39m   [34mE:[39m 9            [34mI:[39m 1,000        [34mR:[39m 163.3        [34mS:[39m 0.0          [34mT:[39m 0:00:24
Pretraining agent on seed data...